### PR TITLE
add support for Sagan

### PR DESCRIPTION
**Pull request by VVelox**
This adds in support for monitoring Sagan.

A few screen shots...

https://i.imgur.com/czE2a9s.png
https://i.imgur.com/upG97Ue.png
https://i.imgur.com/7CTtSgT.png
https://i.imgur.com/T9gsCxq.png

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by VVelox** [comment]
Now depends on https://github.com/librenms/librenms/pull/14087

File: includes/html/pages/device/apps/sagan.inc.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'tab'    => 'apps',
+    'app'    => 'sagan',
+];
+
+print_optionbar_start();
+
+echo generate_link('Totals', $link_array);
+echo '| Instances:';
+$int_int = 0;
+while (isset($sagan_instances[$int_int])) {
```
Comment (murrant) [inline includes/html/pages/device/apps/sagan.inc.php:21]: Replace this while loop with a foreach as I did in the other applications that used it.

Comment (VVelox) [reply includes/html/pages/device/apps/sagan.inc.php]: Done! And more.

**Review by murrant — DISMISSED** [review]

File: doc/Extensions/Applications.md [hunk 2]
```diff
@@ -2179,6 +2179,71 @@ extend supervisord /etc/snmp/supervisord.py
...
 ```
 
+## Sagan
+
+For metrics the stats are migrated as below from the stats JSON.
+
+'f_drop_percent' and 'drop_percent' are computed based on the found data.
+
+```
+uptime              => $json->{stats}{uptime},
```
Comment (murrant) [inline doc/Extensions/Applications.md:2189]: This format is very odd and a little confusing.

Are you saying the application output json and you are changing that json to a different format?  Perhaps that would be better included in the server side script as a comment.

Comment (VVelox) [reply doc/Extensions/Applications.md:2189]: Yeah, basically just copied it directly from the perl code for the extend.

I'll update it to the manner used by jq.

Comment (VVelox) [reply doc/Extensions/Applications.md]: @murrant  And done! That should make it clear now. :)

**Review by murrant — APPROVED** [review]

**Pull request merged**